swarm/storage/ldbstore.go

```diff
@@ -659,38 +761,26 @@ func (s *LDBStore) writeCurrentBatch() error {
 	if l == 0 {
 		return nil
 	}
-	e := s.entryCnt
-	d := s.dataIdx
-	a := s.accessCnt
 	s.batch = newBatch()
-	b.err = s.writeBatch(b, e, d, a)
+	b.err = s.writeBatch(b, wEntryCnt|wAccessCnt|wIndexCnt)
 	close(b.c)
-	for e > s.capacity {
-		log.Trace("for >", "e", e, "s.capacity", s.capacity)
-		// Collect garbage in a separate goroutine
-		// to be able to interrupt this loop by s.quit.
-		done := make(chan struct{})
-		go func() {
-			s.collectGarbage(gcArrayFreeRatio)
-			log.Trace("collectGarbage closing done")
-			close(done)
-		}()
-
-		select {
-		case <-s.quit:
-			return errors.New("CollectGarbage terminated due to quit")
-		case <-done:
-		}
-		e = s.entryCnt
+	if s.entryCnt >= s.capacity {
+		go s.collectGarbage()
 	}
 	return nil
 }
 
 // must be called non concurrently
-func (s *LDBStore) writeBatch(b *dbBatch, entryCnt, dataIdx, accessCnt uint64) error {
-	b.Put(keyEntryCnt, U64ToBytes(entryCnt))
-	b.Put(keyDataIdx, U64ToBytes(dataIdx))
-	b.Put(keyAccessCnt, U64ToBytes(accessCnt))
+func (s *LDBStore) writeBatch(b *dbBatch, wFlag uint8) error {
```

Comment:
do we need this code complexity just to save writing two values in a batch write on each round of GC??? 
i would get rid of it

Reply:
I don't think any superfluous writes are desirable.

I'm not sure why you characterize a couple of bitflags as complex. To me it's very clear what they're doing. Sometimes it feels like the mere existence of bitflags in the code warrants characterizing it as complex :/

An option is having a put of `entryCnt` alone in the `collectGarbage` method. But I find that less readable, to be honest.

Reply:
My opinion is avoid superfluous writes. Let's have a second opinion.